services/api/service.go

```diff
@@ -574,6 +613,86 @@ func (api *RelayAPI) startValidatorRegistrationDBProcessor() {
 	}
 }
 
+// TODO - come up with better name? state interference is not really descriptive name
+func (api *RelayAPI) StateInterferenceChecks(trace *common.CallTrace) (bool, error) {
+	if api.opts.EthNetDetails.Name == "custom" {
+		return api.IsTxWEthDaiSwap(trace)
+	}
+
+	return false, fmt.Errorf("state interference checks not implemented for %s", api.opts.EthNetDetails.Name)
+}
+
+// just check if it goes to the DaiWethPair with a swap tx
```

Comment:
super dumb question: cant we just check the `to` address and compare it ?

Reply:
we could! but that would only cover simpler txs! If someone is doing a DaiWeth swap through one-inch or some dex aggregator then we cannot accommodate those txs!